lib/services/local/okta.go

```diff
@@ -185,38 +164,29 @@ func (o *OktaService) UpdateOktaAssignment(ctx context.Context, assignment types
 // status is a valid transition. If a transition is invalid, it will be logged and the rest of the action statuses
 // will be updated if possible.
 func (o *OktaService) UpdateOktaAssignmentActionStatuses(ctx context.Context, name, status string) (types.OktaAssignment, error) {
-	var assignment types.OktaAssignment
-	err := o.assignmentSvc.RunWhileLocked(ctx, oktaAssignmentModifyLock, oktaAssignmentModifyLockTTL, func(ctx context.Context, b backend.Backend) error {
-		var err error
-		assignment, err = o.assignmentSvc.GetResource(ctx, name)
-		if err != nil {
-			return trace.Wrap(err)
-		}
-
+	var previousAssignment types.OktaAssignment
+	err := o.assignmentSvc.UpdateAndSwapResource(ctx, name, func(assignment types.OktaAssignment) error {
+		previousAssignment = assignment.Copy()
 		for _, action := range assignment.GetActions() {
```

Comment:
Does `GetActions` return a mutable slice, or do we need to then do an `assignment.SetActions` after modifying the actions' statuses?

Same goes for `UpdateOktaAssignment`.

Reply:
It does return a mutable slice.